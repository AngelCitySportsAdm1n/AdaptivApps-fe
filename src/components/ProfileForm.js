import React, { useEffect, useState } from 'react';
import axios from 'axios';
import { connect } from "react-redux";
import Select from 'react-select';
import "../App.css"

const ProfileForm = props => {
    const [userInfo, setUserInfo] = useState({
        disabilities: "",
        type: "",
        state: "",
        city: "",
    });

    console.log(userInfo.disabilities)
    const handleChange = e => {
        let name = e.target.name;
        setUserInfo({ ...userInfo, [name]: e.target.value });
    };
    const handleSelect = selected => {
        setUserInfo({ ...userInfo, disabilities: selected.value })
    }
    const handleStateSelect = selected => {
        setUserInfo({ ...userInfo, state: selected.value })
    }
    const handleTypeSelect = selected => {
        setUserInfo({ ...userInfo, type: selected.value })
    }
    const handleEventSelect = selected => {
        setUserInfo({ ...userInfo, event: selected.value })
    }
    const handleImpairmentSelect = selected => {
        setUserInfo({ ...userInfo, impairment: selected.value })
    }
    console.log(userInfo)


    //array of disabilities for the select
    const disabilities = [
        { label: "Athetosis", value: "Athetosis" },
        { label: "Impaired Muscle Power", value: "Impaired Muscle Power" },
        { label: "Impaired Passive Range of Movement", value: "Impaired Passive Range of Movement" },
        { label: "Limb Deficiency", value: "Limb Deficiency" },
        { label: "Leg Length Difference", value: "Leg Length Difference" },
        { label: "Visual Impairment", value: "Vision" },
        { label: "Short Stature", value: "Short Stature" },
        { label: "Hypertonia", value: "Hypertonia" },
        { label: "Ataxia", value: "Ataxia" },
        { label: "Intellectual Impairment", value: "Intellectual Impairment" },
        { label: "None", Value: "None" }
    ]
    //array of states for the state select
    const states = [
        {
            value: "Alabama",
            label: "AL"
        },
        {
            value: "Alaska",
            label: "AK"
        },
        {
            value: "American Samoa",
            label: "AS"
        },
        {
            value: "Arizona",
            label: "AZ"
        },
        {
            value: "Arkansas",
            label: "AR"
        },
        {
            value: "California",
            label: "CA"
        },
        {
            value: "Colorado",
            label: "CO"
        },
        {
            value: "Connecticut",
            label: "CT"
        },
        {
            value: "Delaware",
            label: "DE"
        },
        {
            value: "District Of Columbia",
            label: "DC"
        },
        {
            value: "Federated States Of Micronesia",
            label: "FM"
        },
        {
            value: "Florida",
            label: "FL"
        },
        {
            value: "Georgia",
            label: "GA"
        },
        {
            value: "Guam",
            label: "GU"
        },
        {
            value: "Hawaii",
            label: "HI"
        },
        {
            value: "Idaho",
            label: "ID"
        },
        {
            value: "Illinois",
            label: "IL"
        },
        {
            value: "Indiana",
            label: "IN"
        },
        {
            value: "Iowa",
            label: "IA"
        },
        {
            value: "Kansas",
            label: "KS"
        },
        {
            value: "Kentucky",
            label: "KY"
        },
        {
            value: "Louisiana",
            label: "LA"
        },
        {
            value: "Maine",
            label: "ME"
        },
        {
            value: "Marshall Islands",
            label: "MH"
        },
        {
            value: "Maryland",
            label: "MD"
        },
        {
            value: "Massachusetts",
            label: "MA"
        },
        {
            value: "Michigan",
            label: "MI"
        },
        {
            value: "Minnesota",
            label: "MN"
        },
        {
            value: "Mississippi",
            label: "MS"
        },
        {
            value: "Missouri",
            label: "MO"
        },
        {
            value: "Montana",
            label: "MT"
        },
        {
            value: "Nebraska",
            label: "NE"
        },
        {
            value: "Nevada",
            label: "NV"
        },
        {
            value: "New Hampshire",
            label: "NH"
        },
        {
            value: "New Jersey",
            label: "NJ"
        },
        {
            value: "New Mexico",
            label: "NM"
        },
        {
            value: "New York",
            label: "NY"
        },
        {
            value: "North Carolina",
            label: "NC"
        },
        {
            value: "North Dakota",
            label: "ND"
        },
        {
            value: "Northern Mariana Islands",
            label: "MP"
        },
        {
            value: "Ohio",
            label: "OH"
        },
        {
            value: "Oklahoma",
            label: "OK"
        },
        {
            value: "Oregon",
            label: "OR"
        },
        {
            value: "Palau",
            label: "PW"
        },
        {
            value: "Pennsylvania",
            label: "PA"
        },
        {
            value: "Puerto Rico",
            label: "PR"
        },
        {
            value: "Rhode Island",
            label: "RI"
        },
        {
            value: "South Carolina",
            label: "SC"
        },
        {
            value: "South Dakota",
            label: "SD"
        },
        {
            value: "Tennessee",
            label: "TN"
        },
        {
            value: "Texas",
            label: "TX"
        },
        {
            value: "Utah",
            label: "UT"
        },
        {
            value: "Vermont",
            label: "VT"
        },
        {
            value: "Virgin Islands",
            label: "VI"
        },
        {
            value: "Virginia",
            label: "VA"
        },
        {
            value: "Washington",
            label: "WA"
        },
        {
            value: "West Virginia",
            label: "WV"
        },
        {
            value: "Wisconsin",
            label: "WI"
        },
        {
            value: "Wyoming",
            label: "WY"
        }
    ];
    //array of user types
    const userType = [
        { label: "Coach", value: "Coach" },
        { label: "Athlete", value: "Athlete" },
        { label: "Volunteer", value: "Volunteer" }
    ]
    //array of events
    const events = [

    ]
    const impairment =
    [
    { label: "ALS",  value: "ALS" },
    { label: "Amputation / Limb Difference",  value: "Amputation / Limb Difference" },
    { label: "Arthrogryposis" ,  value: "Arthrogryposis" },
    { label: "Ataxia",  value: "Ataxia" },
    { label: "Blind / Visual Impairment",  value: "Blind / Visual Impairment" },
    { label: "Brachial Plexus Injury",  value: "Brachial Plexus Injury" },
    { label: "Cauda Equina syndrome",  value: "Cauda Equina syndrome" },
    { label: " Cerebal Palsy",  value: " Cerebal Palsy" },
    { label: "Chronic joint immobilization / Arthritis" ,  value: "Chronic joint immobilization / Arthritis" },
    { label: "CMT - Charcot - Maire - Tooth",  value: "CMT - Charcot - Maire - Tooth" },
    { label: "Drop Foot",  value: "Drop Foot" },
    { label: "Dwarfism",  value:  "Dwarfism" },
    { label: " Ehlers Danos Syndrom",  value: " Ehlers Danos Syndrom" },
    { label: "Fibromyalgia",  value: "Fibromyalgia" },
    { label: "Guillain–Barré syndrome",  value:"Guillain–Barré syndrome" },
    { label: "Leg Length Difference",  value: "Leg Length Difference" },
    { label: "Multiple Sclerosis",  value: "Multiple Sclerosis" },
    { label: "Muscular Dystrophy",   value: "Muscular Dystrophy" },
    { label: "Osteogenesis imperfecta",  value: "Osteogenesis imperfecta" },
    { label: "Osteoporosis",  value: "Osteoporosis" },
    { label: "Parkinsons",  value: "Parkinsons" },
    { label: "Polio",  value: "Polio" },
    { label: "Prader - Willi Syndrome(PWS)",  value: "Prader - Willi Syndrome(PWS)" },
    { label: "Short Stature",  value: "Short Stature" },
    { label: "Spina Bifida",  value: "Spina Bifida" },
    { label: "Spinal Cord Injury - Paraplegia" ,  value: "Spinal Cord Injury - Paraplegia" },
    { label: "Spinal Cord Injury - Quadraplegia" ,  value: "Spinal Cord Injury - Quadraplegia" },
    { label: "Stroke" ,  value: "Stroke" },
    { label: "TBI" ,  value: "TBI" },
    { label: "Other",  value: "Other" },
    ]
    return (
        <form>

            <label>Disability</label>
            <input
                type="text"
                name=""
                placeholder=""
                value={userInfo.disabilities}
                onChange={handleChange}
            />


            <label>State</label>
            <Select options={states}
                name="disabilities"
                value={userInfo.state}
                onChange={handleStateSelect}
                placeholder={userInfo.state} />

            <label>City</label>
            <input
                type="text"
                name="city"
                placeholder=""
                value={userInfo.city}
                onChange={handleChange}
            />

            <label>User Type</label>
            <Select options={userType}
                name="type"
                value={userInfo.type}
                onChange={handleTypeSelect}
                placeholder={userInfo.type} />

            {/* Can use is multi to give people multiple options not required though. */}
            {/* made it so certain things only display depending on the user type */}
            <div className={userInfo.type == "Athlete" ? "athlete" : "none"}>
                <label>Disability Group</label>
                <Select options={disabilities}
                    name="disabilities"
                    value={userInfo.disabilities}
                    onChange={handleSelect}
                    placeholder={userInfo.disabilities} />

                <label>Specific Disability</label>

                <Select options={impairment}
                    name="impairment"
                    value={userInfo.impairment}
                    onChange={handleImpairmentSelect}
                    placeholder={userInfo.disabilities} />
            </div>

        </form>
    )
}

const mapStateToProps = state => {
    return state;
};


export default connect(
    mapStateToProps)
    (ProfileForm);