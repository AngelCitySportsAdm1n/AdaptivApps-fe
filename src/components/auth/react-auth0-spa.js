import React, { useState, useEffect, useContext } from 'react';
import createAuth0Client from '@auth0/auth0-spa-js';
import PropTypes from 'prop-types';

const DEFAULT_REDIRECT_CALLBACK = () =>
  window.history.replaceState({}, document.title, window.location.pathname);

export const Auth0Context = React.createContext();
export const useAuth0 = () => useContext(Auth0Context);
//console.log('console.log useAuth0', useAuth0.getTokenSilently);
export const Auth0Provider = ({
  children,
  onRedirectCallback = DEFAULT_REDIRECT_CALLBACK,
  ...initOptions
}) => {
  const [isAuthenticated, setIsAuthenticated] = useState();
  const [user, setUser] = useState();
  const [auth0Client, setAuth0] = useState();
  const [loading, setLoading] = useState(true);
  const [popupOpen, setPopupOpen] = useState(false);

  useEffect(() => {
    const initAuth0 = async () => {
      const auth0FromHook = await createAuth0Client(initOptions);
      setAuth0(auth0FromHook);

      if (
        window.location.search.includes('code=') &&
        window.location.search.includes('state=')
      ) {
        const { appState } = await auth0FromHook.handleRedirectCallback();
        onRedirectCallback(appState);
      }
      const tokenP = auth0FromHook.getIdTokenClaims();

      const isAuthenticated = await auth0FromHook.isAuthenticated();
      //console.log(
      //'Showing this auth0FrameHook',
      //   auth0FromHook.getIdTokenClaims()
      // );
      // checks to make sure user is authenticated and sets user if true.
      setIsAuthenticated(isAuthenticated);

      if (isAuthenticated) {
        const token = await tokenP;
        //console.log('Showing tokenP', token.__raw);
        //localStorage.setItem('golden', token.__raw);
        const user = await auth0FromHook.getUser();
        setUser(user);
        //console.log('authenticated', user);
      }

      setLoading(false);
    };
    initAuth0();
    // eslint-disable-next-line
  }, []);

  const loginWithPopup = async (params = {}) => {
    setPopupOpen(true);
    try {
      await auth0Client.loginWithPopup(params);
    } catch (error) {
      //console.error(error);
    } finally {
      setPopupOpen(false);
    }
    const user = await auth0Client.getUser();
    setUser(user);
    setIsAuthenticated(true);
    //console.log('user', user);
  };

  const handleRedirectCallback = async () => {
    setLoading(true);
    await auth0Client.handleRedirectCallback();
    const user = await auth0Client.getUser();
    setLoading(false);
    setIsAuthenticated(true);
    setUser(user);
    //console.log('redirect', user);
  };
  return (
    <Auth0Context.Provider
      value={{
        isAuthenticated,
        user,
        loading,
        popupOpen,
        loginWithPopup,
        handleRedirectCallback,
        getIdTokenClaims: (...p) => auth0Client.getIdTokenClaims(...p),
        loginWithRedirect: (...p) => auth0Client.loginWithRedirect(...p),
        getTokenSilently: (...p) => auth0Client.getTokenSilently(...p),
        getTokenWithPopup: (...p) => auth0Client.getTokenWithPopup(...p),
        logout: (...p) => auth0Client.logout(...p),
      }}
    >
      {children}
    </Auth0Context.Provider>
  );
};

Auth0Provider.propTypes = {
  children: PropTypes.any,
  onRedirectCallback: PropTypes.any,
};
