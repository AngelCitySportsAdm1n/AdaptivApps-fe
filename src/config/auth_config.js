module.exports = {
  domain: process.env.REACT_APP_CLIENT_DOMAIN,
  clientId: process.env.REACT_APP_CLIENT_ID,
  roleUrl: 'http://adaptivapps.com/roles',
  audience: process.env.REACT_APP_AUDIENCE,
  responseType: 'token id_token',
  scope: 'openid email',
};
