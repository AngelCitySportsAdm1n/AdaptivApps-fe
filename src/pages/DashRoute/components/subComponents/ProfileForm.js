import React, { useState } from 'react';
import config from '../../../../components/auth/auth_config';
import { useForm } from 'react-hook-form';
import {
  Flex,
  Box,
  Text,
  Button,
  Form,
  Input,
  theme,
  TextArea,
  Select,
} from 'adaptiv-ui';
import PropTypes from 'prop-types';

// This is the form being used in UserDashboard
const ProfileForm = ({ profile, user, updateProfile }) => {
  const [updated, setUpdated] = useState(false)

  const { handleSubmit, register } = useForm({
    mode: 'onChange',
    defaultValues: {
      email: user?.email,
      firstName: profile?.firstName,
      lastName: profile?.lastName,
      displayName: profile?.displayName,
      birthday: profile?.birthday,
      bio: profile?.bio,
      disability: profile?.disability
    },
  });

  const onSubmit = (formValues, e) => {
    e.preventDefault()
    updateProfile({
      variables: {
        email: user.email,
        firstName: formValues.firstName === '' ? profile.firstName : formValues.firstName,
        lastName: formValues.lastName === '' ? profile.lastName : formValues.lastName,
        displayName: formValues.displayName === '' ? profile.displayName : formValues.displayName,
        birthday: formValues.birthday === '' ? profile.birthday : formValues.birthday,
        bio: formValues.bio === '' ? profile.bio : formValues.bio,
        disability: formValues.disability === '' ? profile.disability : formValues.disability,
        legal: formValues.legal === '' ? profile.legal : formValues.legal
      },
    });
  };
  
  return (
    <Flex ai_start col stretch>
      <Text xlf bold mm>
        Account Information
      </Text>
      <Box h="0.2rem" w="90%" bg="lightgrey" />
      <Flex mm col ai_start>
        <Flex jc_between ai_center>
          <Box sqr="5rem">
            <img src={user.picture} alt="Profile" />
          </Box>
          <Text lf sm>
            {profile && profile.firstName !== null
              ? `${profile && profile.firstName} ${profile && profile.lastName}`
              : user.name}{' '}
            {user[config.roleUrl].includes('Admin') ? (
              <Text>{user[config.roleUrl]}</Text>
            ) : null}
          </Text>
        </Flex>

        <Box h="1rem" />

        <Flex ai_start col>
          <Text lf bold>
            Account Email Address
          </Text>

          <Flex ai_center>
            <Text lf>{profile ? profile.email : user.email}</Text>
            <Button
              primary
              jc_center
              mm
              border={`2px solid ${theme.primary}`}
              w="7.75rem"
              h="3.75rem"
              aria-label="Change email for this user"
              onClick={() => alert('functionality coming next release canvas')}
            >
              Change
            </Button>
          </Flex>
        </Flex>

        <Form ai_start col stretch onSubmit={handleSubmit(onSubmit)}>
          <Text xlf bold>
            Personal Information
          </Text>

          <Box h="1rem" />

          <Flex jc_between stretch>
            <Flex ai_start col>
              <Text mf>First Name</Text>
              <Flex ai_center>
                <Input
                  type="text"
                  placeholder={profile ? profile.firstName : null}
                  w="25rem"
                  name="firstName"
                  ref={register}
                />
              </Flex>
            </Flex>

            <Flex ai_start col>
              <Text mf>Last Name</Text>
              <Flex ai_center>
                <Input
                  type="text"
                  placeholder={profile ? profile.lastName : null}
                  w="25rem"
                  name="lastName"
                  ref={register}
                />
              </Flex>
            </Flex>
          </Flex>

          <Flex jc_between stretch>
            <Flex ai_start col>
              <Text mf>Display Name</Text>
              <Flex ai_center>
                <Input
                  type="text"
                  placeholder={profile ? profile.displayName : null}
                  w="25rem"
                  name="displayName"
                  ref={register}
                />
              </Flex>
            </Flex>

            <Flex ai_start col>
              <Text mf>Date of Birth</Text>
              <Flex ai_center>
                <Input type="text" w="25rem" name="birthday" ref={register} placeholder={profile ? profile.birthday : 'mm/dd/yyyy'} />
              </Flex>
            </Flex>
          </Flex>

          <Flex ai_start col stretch>
            <Text mf>Bio</Text>
            <TextArea rows="8" cols="60" name="bio" ref={register} placeholder={profile ? profile.bio : null} />
          </Flex>

          <Flex jc_between stretch>
            <Flex ai_start col>
              <Text mf>Disability Status</Text>
              <Flex ai_center>
                <Input
                  type="select"
                  w="25rem"
                  name="disability"
                  ref={register}
                  placeholder={profile ? profile.disability : null}
                />
              </Flex>
            </Flex>

            <Flex ai_start col>
              <Text mf>Are you over 18 years old?</Text>
              <Flex ai_center>
                <Select w="20rem" name="legal" ref={register}>
                  <option value={true}>Yes</option>
                  <option value={false}>No</option>
                </Select>
              </Flex>
            </Flex>
          </Flex>

          <Flex w='50%' jc_between ai_center>
            <Button
              type="submit"
              jc_center
              primary
              border={`2px solid ${theme.primary}`}
              w="9rem"
              h="4rem"
              aria-label="save changes to user profile"
              onClick={() => {
                setUpdated(true);
              }}
            >
              Save
            </Button>
            {updated === true ? <Text>Profile updated!</Text> : null}
          </Flex>

        </Form>
      </Flex>
    </Flex>
  );
};

export default ProfileForm;

ProfileForm.propTypes = {
  profile: PropTypes.object,
  user: PropTypes.object,
};
