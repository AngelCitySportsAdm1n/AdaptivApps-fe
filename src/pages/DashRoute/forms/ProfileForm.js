import React from 'react';
import config from '../../../components/auth/auth_config';
import { useForm } from 'react-hook-form';
import {
  Flex,
  Box,
  Text,
  Button,
  Form,
  Input,
  theme,
  Select,
  TextArea,
} from 'adaptiv-ui';
import PropTypes from 'prop-types';

// Form for users to fill out their personal info and update profile
const UserDashboard = props => {
  const { profile, user } = props;
  const { handleSubmit, register } = useForm();

  const onSubmit = () => {
    alert('functionality coming next release canvas');
  };

  return (
    <Flex ai_start col stretch>
      <Text xlf bold mm>
        Account Information
      </Text>
      <Box h="0.2rem" w="90%" bg="lightgrey" />
      <Flex mm col ai_start>
        <Flex jc_between ai_center>
          <Box sqr="5rem">
            <img src={user.picture} alt="Profile" />
          </Box>
          <Text lf sm>
            {profile ? profile.firstName : null}{' '}
            {profile ? profile.lastName : null} ({user[config.roleUrl]})
          </Text>
        </Flex>

        <Box h="2rem" />

        <Flex ai_start col>
          <Text lf bold>
            Account Email Address
          </Text>

          <Flex ai_center>
            <Text lf>{profile ? profile.email : user.email}</Text>
            <Button
              primary
              jc_center
              mm
              border={`2px solid ${theme.primary}`}
              w="7.75rem"
              h="3.75rem"
              aria-label="Change email for this user"
              onClick={() => alert('functionality coming next release canvas')}
            >
              Change
            </Button>
          </Flex>
        </Flex>

        <Form ai_start col stretch onSubmit={handleSubmit(onSubmit)}>
          <Text xlf bold>
            Personal Information
          </Text>

          <Box h="2rem" />

          <Flex jc_between stretch>
            <Flex ai_start col>
              <Text mf>First Name</Text>
              <Flex ai_center>
                <Input
                  type="text"
                  placeholder={profile ? profile.firstName : null}
                  w="25rem"
                  name="firstName"
                  ref={register}
                />
              </Flex>
            </Flex>

            <Flex ai_start col>
              <Text mf>Last Name</Text>
              <Flex ai_center>
                <Input
                  type="text"
                  placeholder={profile ? profile.lastName : null}
                  w="25rem"
                  name="lastName"
                  ref={register}
                />
              </Flex>
            </Flex>
          </Flex>

          <Flex jc_between stretch>
            <Flex ai_start col>
              <Text mf>Display Name</Text>
              <Flex ai_center>
                <Input
                  type="text"
                  placeholder={profile ? profile.displayName : null}
                  w="25rem"
                  name="displayName"
                  ref={register}
                />
              </Flex>
            </Flex>

            <Flex ai_start col>
              <Text mf>Date of Birth</Text>
              <Flex ai_center>
                <Input type="date" w="25rem" name="birthday" ref={register} />
              </Flex>
            </Flex>
          </Flex>

          <Flex ai_start col stretch>
            <Text mf>Bio</Text>
            <TextArea rows="8" cols="60" name="bio" ref={register} />
          </Flex>

          <Flex jc_between stretch>
            <Flex ai_start col>
              <Text mf>Disability Status</Text>
              <Flex ai_center>
                <Input
                  type="select"
                  placeholder=""
                  w="25rem"
                  name="disability"
                  ref={register}
                />
              </Flex>
            </Flex>

            <Flex ai_start col>
              <Text mf>Are you over 18 years old?</Text>
              <Flex ai_center>
                <Select w="20rem" name="legal" ref={register}>
                  <option value={true}>Yes</option>
                  <option value={false}>No</option>
                </Select>
              </Flex>
            </Flex>
          </Flex>

          <Button
            type="submit"
            jc_center
            primary
            border={`2px solid ${theme.primary}`}
            w="9rem"
            h="4rem"
            aria-label="save changes to user profile"
          >
            Save
          </Button>
        </Form>
      </Flex>
    </Flex>
  );
};

export default UserDashboard;

UserDashboard.propTypes = {
  profile: PropTypes.object,
  user: PropTypes.object,
};
