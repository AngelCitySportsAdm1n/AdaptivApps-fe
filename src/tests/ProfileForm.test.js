import React from 'react';
import ReactDom from 'react-dom';
// import { done } from '@testing-library/react';
import { asyncAuth } from '../config/react-auth0-spa'
import ProfileForm from '../pages/UserProfile/ProfileForm';
import { getByText } from '@testing-library/react';

test('renders app', async () => {
  const auth0 = await asyncAuth;
  if (auth0 === true) {
    const div = document.createElement('div');
    ReactDom.render(<ProfileForm />, div);
  }
});

test('renders first name field', async () => {
  const auth0 = await asyncAuth;
  if (auth0 === true) {
    ReactDom.render(<ProfileForm />)
    getByText(/there/i)
  }
})