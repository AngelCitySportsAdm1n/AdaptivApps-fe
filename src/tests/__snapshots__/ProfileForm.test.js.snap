// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders profile form correctly 1`] = `
<div
  className="sc-ifAKCX cmbMjk"
>
  <h4
    style={
      Object {
        "fontSize": "2.4rem",
        "marginBottom": "0.5rem",
      }
    }
  >
    Account Information
  </h4>
  <div
    className="sc-iwsKbI jOijdN"
  />
  <div
    className="sc-ifAKCX gDJgfb"
    style={
      Object {
        "marginTop": 0,
      }
    }
  >
    <div
      className="sc-ifAKCX gxpeBj"
      style={
        Object {
          "marginTop": 0,
        }
      }
    >
      <div
        className="sc-iwsKbI dcswkN"
      >
        <img
          alt="Profile"
        />
      </div>
      <div
        className="sc-gqjmRU bQOFXJ"
      >
         
      </div>
    </div>
    <div
      className="sc-iwsKbI bqkBdz"
    />
    <div
      className="sc-ifAKCX gHfEQj"
    >
      <h5
        style={
          Object {
            "fontSize": "2.1rem",
            "marginLeft": 0,
          }
        }
      >
        Account Email Address
      </h5>
      <div
        className="sc-ifAKCX jLfGdD"
      >
        <div
          className="sc-gqjmRU criljC"
        />
      </div>
      <div
        className="sc-iwsKbI bqkBdz"
      />
    </div>
    <form
      className="sc-EHOje gLPZAQ"
      onSubmit={[Function]}
    >
      <h5
        style={
          Object {
            "fontSize": "2.1rem",
            "marginLeft": 0,
          }
        }
      >
        Personal Information
      </h5>
      <div
        className="sc-ifAKCX euqiKW"
      >
        <div
          className="sc-ifAKCX gHfEQj"
        >
          <label
            className="sc-cSHVUG joCGnZ"
            htmlFor="firstName"
            style={
              Object {
                "marginBottom": "0.2rem",
              }
            }
          >
            First Name
          </label>
          <div
            className="sc-ifAKCX jLfGdD"
          >
            <input
              autoFocus={true}
              className="sc-bZQynM dDlBhs"
              id="firstName"
              name="firstName"
              placeholder={null}
              style={
                Object {
                  "marginLeft": 0,
                  "marginTop": 0,
                }
              }
              type="text"
            />
          </div>
        </div>
      </div>
      <div
        className="sc-ifAKCX euqiKW"
      >
        <div
          className="sc-ifAKCX gHfEQj"
        >
          <label
            className="sc-cSHVUG joCGnZ"
            htmlFor="displayName"
            style={
              Object {
                "marginBottom": "0.2rem",
              }
            }
          >
            Display Name
          </label>
          <div
            className="sc-ifAKCX jLfGdD"
          >
            <input
              className="sc-bZQynM dDlBhs"
              id="displayName"
              name="displayName"
              placeholder={null}
              style={
                Object {
                  "marginLeft": 0,
                  "marginTop": 0,
                }
              }
              type="text"
            />
          </div>
        </div>
        <div
          className="sc-ifAKCX gHfEQj"
        >
          <label
            className="sc-cSHVUG joCGnZ"
            htmlFor="birthday"
            style={
              Object {
                "marginBottom": "0.2rem",
              }
            }
          >
            Date of Birth
          </label>
          <div
            className="sc-ifAKCX jLfGdD"
          >
            <input
              className="sc-bZQynM dDlBhs"
              id="birthday"
              name="birthday"
              placeholder="mm/dd/yyyy"
              style={
                Object {
                  "marginLeft": 0,
                  "marginTop": 0,
                }
              }
              type="text"
            />
          </div>
        </div>
      </div>
      <div
        className="sc-ifAKCX cmbMjk"
      >
        <label
          className="sc-cSHVUG joCGnZ"
          htmlFor="bio"
          style={
            Object {
              "marginBottom": "0.2rem",
            }
          }
        >
          Bio
        </label>
        <textarea
          className="sc-fjdhpX cfOWcZ"
          cols="60"
          id="bio"
          name="bio"
          placeholder={null}
          rows="8"
          style={
            Object {
              "marginLeft": 0,
              "marginTop": 0,
            }
          }
        />
      </div>
      <div
        className="sc-ifAKCX euqiKW"
      >
        <div
          className="sc-ifAKCX gHfEQj"
        >
          <label
            htmlFor="disability"
            style={
              Object {
                "marginBottom": "0.2rem",
              }
            }
          >
            Disability Status
          </label>
          <div
            className="sc-ifAKCX jLfGdD"
          >
            <input
              className="sc-bZQynM dDlBhs"
              id="disability"
              name="disability"
              placeholder={null}
              style={
                Object {
                  "marginLeft": 0,
                  "marginTop": 0,
                }
              }
              type="select"
            />
          </div>
        </div>
        <div
          className="sc-ifAKCX gHfEQj"
        >
          <label
            htmlFor="legal"
            style={
              Object {
                "marginBottom": "0.2rem",
              }
            }
          >
            Are you over 18 years old?
          </label>
          <div
            className="sc-ifAKCX jLfGdD"
          >
            <select
              className="sc-VigVT iDFudI"
              id="legal"
              name="legal"
              style={
                Object {
                  "marginLeft": 0,
                  "marginTop": 0,
                }
              }
            >
              <option
                value={true}
              >
                Yes
              </option>
              <option
                value={false}
              >
                No
              </option>
            </select>
          </div>
        </div>
      </div>
      <div
        className="sc-ifAKCX jTYOJY"
      >
        <button
          aria-label="save changes to user profile"
          className="sc-gzVnrw zKOMm"
          onClick={[Function]}
          style={
            Object {
              "marginLeft": 0,
            }
          }
          type="submit"
        >
          Save
        </button>
      </div>
    </form>
  </div>
</div>
`;
